csrc/ir/utils.cpp

```diff
@@ -769,6 +769,18 @@ bool isIndexedConsumerID(const TensorView* tv, const IterDomain* id) {
       tv->definition()->as<ScatterOp>()->getIndexedID() == id;
 }
 
+TensorView* getIndexSelectOutputTv(const TensorView* tv) {
+  for (auto expr : tv->uses()) {
+    if (expr->isA<IndexSelectOp>()) {
+      auto idx_sel = expr->as<IndexSelectOp>();
+      if (idx_sel->input(0) == tv) {
```

Comment:
this line contradicts the function comment. We are only looking for IndexSelectOp that consumes tv as the lookup tv.